quickfixj-core/src/test/java/quickfix/test/acceptance/ATApplication.java

```diff
@@ -81,6 +83,11 @@ public void toApp(Message message, SessionID sessionID) throws DoNotSend {
     public void fromAdmin(Message message, SessionID sessionID) throws FieldNotFound,
             IncorrectDataFormat, IncorrectTagValue, RejectLogon {
         assertNoSessionLock(sessionID);
+
+        if (count > 2) { // leave the logon and some heartbeats go through, then logout the session.
+            Session.lookupSession(sessionID).logout();
```

Comment:
As communicated using `logout()` will disable the session until `logon()` is called again. Please use `generateLogout()` or throw `RejectLogon`.